website/addons/figshare/views/config.py

```diff
@@ -38,16 +38,31 @@ def figshare_config_put(node_addon, auth, **kwargs):
     fields = request.json.get('selected', {})
     node = node_addon.owner
 
-    folder = {
-        'name': fields.get('name') or '',
-        'id': fields.get('id') or None,
-        'type': fields.get('type') or None,
-    }
-    node_addon.update_fields(folder, node, auth)
+    name = fields.get('name')
+    figshare_id = fields.get('id')
+    figshare_type = fields.get('type')
+
+    if not name or not figshare_id or not figshare_type:
+        return HTTPError(http.BAD_REQUEST, message="""
```

Comment:
You must raise the HTTPError, not return it.

Also, in order for the message to be included in the response body, you need to do:

```python
raise HTTPError(http.BAD_REQUEST, data=dict(
    message_long='You must provide name, id, and type'
))
```